control/controlclient/noise.go

```diff
@@ -295,7 +295,9 @@ func (nc *NoiseClient) getConn(ctx context.Context) (*noiseConn, error) {
 	}
 	nc.mu.Unlock()
 
-	conn, err, _ := nc.sfDial.Do(struct{}{}, nc.dial)
+	conn, err, _ := nc.sfDial.Do(struct{}{}, func() (*noiseConn, error) {
+		return nc.dial(ctx)
+	})
```

Comment:
this doesn't seem right. If N dials are coalesced together with single flight, N-1 shouldn't be canceled just because 1 of them's context was canceled?

Reply:
fair point, any suggestion on how to handle this then? i.e. how do i signal shutdown?